### PR TITLE
Address #2448

### LINKED ISSUE #2448: [3.16] Support semanticTokens missing methods
[SPEC](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_semanticTokens)
lsp-mode is missing the implementation of:
* :leftwards_arrow_with_hook: `textDocument/semanticTokens/full/delta` witch should improve performance bringing only the tokens relative to existing ones
* :arrow_left: `workspace/semanticTokens/refresh` not critical, which allows server request client to refresh the tokens at any time.

### PR DESCRIPTION
**Pull request by sebastiansturm**
this PR adds support for delta and refresh requests and fixes flickering seen in combination with some underlying highlighting mechanisms (tree-sitter-hl, clojure's syntax highlighter)

Fixes #2448

### CODE REVIEW

File: lsp-semantic-tokens.el [hunk 1]
```diff
@@ -44,6 +44,23 @@ or modifier type returned by a language server has no face associated with it."
   :group 'lsp-semantic-tokens
   :type 'boolean)
 
+(defcustom lsp-semantic-tokens-allow-ranged-requests t
+  "Whether to use ranged semantic token requests when available.
+
+Note that even when this is set to t, delta requests will
+be preferred whenever possible, unless
+`lsp-semantic-tokens-allow-delta-requests' is false."
+  :group 'lsp-mode
```
Comment (ericdallo) [inline lsp-semantic-tokens.el:53]: Could you change the groups to lsp-semantic-tokens to make those variables documentation show in correct sections of webpage?

**Review by ericdallo — COMMENTED** [review]
Looks good, I'll test manually during this week

**Comment by ericdallo** [comment]
@sebastiansturm I realized I didn't implemented delta support for clojure-lsp yet, I'll use this branch as a base to implement the server

**Comment by ericdallo** [comment]
@sebastiansturm Would you mind merge master in this branch? I intend to test latest changes and this branch this week

**Comment by sebastiansturm** [comment]
> @sebastiansturm Would you mind merge master in this branch? I intend to test latest changes and this branch this week

I just rebased and pushed. Got some error message during fontification with rust-analyzer that I'll have to look into in the following days, but I pushed the current state anyway since semantic tokens seemed to work as usual with clojure-lsp. Thanks for testing!

**Comment by ericdallo** [comment]
@sebastiansturm I tested this PR with clojure-lsp and found no major issues with semantic tokens full and range, I couldn't test delta but this seems to have improvements on semantic-tokens on general, so LGTM to merge, WDYT?

**Comment by sebastiansturm** [comment]
at least I'm not aware of any open issues since the CI failures have been resolved, and I didn't run into problems using it myself. So if you didn't either, then I think it's probably good enough to merge. Thanks for your help!

**Review by ericdallo — APPROVED** [review]

**Pull request merged**